sentry/src/main/java/io/sentry/ISpan.java

```diff
@@ -55,7 +56,7 @@ ISpan startChild(
    */
   @Nullable
   @ApiStatus.Experimental
-  BaggageHeader toBaggageHeader();
+  BaggageHeader toBaggageHeader(@Nullable List<String> thirdPartyBaggageHeaders);
```

Comment:
Not sure if `thirdPartyBaggageHeaders` should be part of the Public API, I'd rather handle it internally.

Reply:
Not sure what you mean. This is used for handing over all existing `baggage` headers that already exist on the outgoing request before we try to add sentry specific values to it. To check for limits and remove sentry entries from the existing headers and replace them with the ones we want we have to pass them in.

Reply:
The problem is that a user when calling `toBaggageHeader` has to pass a `thirdPartyBaggageHeaders` even if they don't know what this is.
This is a good or bad thing I'm unsure, on Dart, I've merged them internally, and I've not exposed the `thirdPartyBaggageHeaders` param as a Public API and that's my point, let's talk in person about this.

Reply:
After discussing I've opened a PR https://github.com/getsentry/develop/pull/689 to mention how Java and Dart handle this case. Both ways of dealing with it are OK at the moment.